src/integ/groovy/org/mastercoin/test/rpc/MSCSendToOwnersSpec.groovy

```diff
@@ -27,21 +27,32 @@ class MSCSendToOwnersSpec extends BaseRegTestSpec {
     def "STO calculates correct fees for the simple case"() {
         setup:
         def startingBTC = 10.0
-        def startingMSC = 1000
-        def amountSent = 100
+        def startingMSC = 1000.0
+        def amountSent = 100.0
         def fundedAddress = createFundedAddress(startingBTC, startingMSC)
         def currencyID = TMSC
+        def expectedBalance = 0.0
 
         when: "We Send to Owners"
-        def startingPropBal = getbalance_MP(fundedAddress, currencyID).balance
+        def startBalances = getallbalancesforid_MP(currencyID)
+        def startBalanceSender = getbalance_MP(fundedAddress, currencyID).balance
+        def numberOfHolders = startBalances.size()
         sendToOwnersMP(fundedAddress, currencyID, amountSent)
 
-        and: "we generate a block"
+        and: "We generate a block"
         generateBlock()
-        def balances = getallbalancesforid_MP(currencyID)
-        def numberDest = balances.size() - 1
-        def totalFee = numberDest * stoFeePerAddress
-        def expectedBalance = (numberDest == 0) ? startingPropBal : startingPropBal - amountSent - totalFee
+
+        // The fee for each receiver is #stoFeePerAddress
+        if (numberOfHolders > 1) {
```

Comment:
Good point. In regtest mode we have full control here, but this isn't necessarily the case on mainnet. What do you suggest? The RPC call `getsto_MP` has a fee field, but it introduces another layer and there was already one flaw in the fee calculation in Core, see https://github.com/mastercoin-MSC/mastercore/pull/271.